PMM-10522 RN for PMM 2.30.0 (#850)
percona/pmm-doc: docs/release-notes/2.30.0.md

        This real-time dashboard contains Oplog details such as Recovery Window, Processing Time, Buffer Capacity, and Oplog Operations. [documentation](https://docs.percona.com/percona-monitoring-and-management/details/dashboards/dashboard-mongodb-experimental_oplog.html).

### Database as a Service (DBaaS)

Enhanced the user experience by simplifying the configuration and deployment of DBaaS database clusters:

- With PMM 2.30.0, you can create a DB cluster literally at the click of a button. All the fields will be automatically populated with the default value. Click *Register*, and your DB cluster is created.

- Starting with PMM 2.30.0, we have simplified the DBaaS configuration and deployment of database clusters by providing suggestions on the k8s cluster name based on the kubeconfig file.

- We have simplified DBaaS and Percona Platform connection configuration by suggesting and automatically setting the public address of the PMM server if it's not set up. This happens when connecting to Percona Platform, adding a Kubernetes (K8s) cluster, or adding a DB cluster.

### Operators support

- PMM now supports Percona Operators MySQL (1.11) and MongoDB (1.11).

> - PMM now supports MongoDB version 5.0.

Reviewer: We don't support MongoDB 5 in dbaas yet https://jira.percona.com/browse/PMM-9556.

# Percona Monitoring and Management 2.30.0

| **Release date:** | Aug 22, 2022                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important"
    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

#### New Experimental dashboards

!!! caution alert alert-warning "Important"

>     These experimental dashboards aresubject to change. It is recommended to use these dashboards for testing purposes only.

Reviewer 2: ```suggestion
    These experimental dashboards are subject to change. It is recommended to use these dashboards for testing purposes only.
```

    - **Experimental MongoDB Collection Overview**
        
        This dashboard contains panels for data about the Hottest Collections in the MongoDB database.

    - **Experimental MongoDB Collection Details**

        This experimental dashboard provides detailed information about the top collections by document count, size, and document read for MongoDB databases.

    - **Experimental MongoDB Oplog Details**

        This real-time dashboard contains Oplog details such as Recovery Window, Processing Time, Buffer Capacity, and Oplog Operations.

### Database as a Service (DBaaS)

Enhanced the user experience by simplifying the configuration and deployment of DBaaS database clusters:

- With PMM 2.30.0, you can create a DB cluster literally at the click of a button. All the fields will be automatically populated with the default value. Click Register, and your DB cluster is created.

- You can easily create a Kubernetes cluster with PMM 2.30.0. When you copy the value of the Kubeconfig file, the Kubernetes Cluster Name gets auto-populated from the contents of the kubeconfig file.

> - Starting with PMM 2.23.0, we have simplified DBaaS configuration and deployment of database clusters by providing suggestions on the k8s cluster name based on the Kubeconfig file.

Reviewer 2: ```suggestion
- Starting with PMM 2.30.0, we have simplified DBaaS configuration and deployment of database clusters by providing suggestions on the k8s cluster name based on the Kubeconfig file.
```

### Operators support

> - PMM now supports Percona Operators MySQL (1.11) and MongoDB (1.12).

Reviewer 2: @beatahandzelova do we have 1.12? I think it  is failing tests and not supported for 2.30.0

Reviewer: Yes MongoDB (1.12) is not supported yet.

> - PMM now provides MongoDB v5 deployment support. 

Reviewer 2: @beatahandzelova this is not true, right?

Reviewer: @rasika-chivate please remove all occurrences of MongoDB (1.12) operator. We support Percona Operators MySQL (1.11) and MongoDB (1.11) at the moment.

Author: ok

## New Features

- [PMM-9652](https://jira.percona.com/browse/PMM-9652): New experimental dashboards for MongoDB are now available with more MongoDB metrics and data like Hottest collections, Data Size, recovery Window, Processing Time, Buffer Capacity, Oplog Operations, and so on.

- [PMM-10367](https://jira.percona.com/browse/PMM-10367): We have introduced a new experimental Home Dashboard in PMM version 2.30.0. The primary goal of this Home dashboard is to improve the performance while monitoring a large number of nodes or services in PMM. Repeated rows (for a large number of nodes) on the dashboard can take a considerable time to load Home Dashboard UI. So, to address this issue, we have developed a very user-friendly and insightful new experimental Home Dashboard.

- [PMM-10327](https://jira.percona.com/browse/PMM-10327):  You can create a free K8s cluster via Percona Platform Portal and use it for testing DBaaS in PMM.
    
    !!! note alert alert-primary ""
        This feature will be available on https://portal.percona.com/ with the PMM 2.30 GA release.

 
## Improvements

> - [PMM-10340](https://jira.percona.com/browse/PMM-10340): DBaas: Enhanced user experience to simplify DBaaS configuration and deployment of Database clusters with suggestions on k8s cluster name based on k8s config file.

Reviewer 2: ```suggestion
- [PMM-10340](https://jira.percona.com/browse/PMM-10340): DBaaS: Enhanced user experience to simplify DBaaS configuration and deployment of Database clusters with suggestions on k8s cluster name based on k8s config file.
```

  
- [PMM-9914](https://jira.percona.com/browse/PMM-9914): PMM generates config files into the `/tmp` folder when a service is added to monitoring. Files in `/tmp` can get deleted by the Operating system. With PMM 2.30.0, we have ensured that the pmm-agent successfully restarts the child agents even though the config files in the /tmp folder are deleted and the scraper/exporter processes are killed thereby ensuring that the exporter is not broken.

    
  
- [PMM-9994](https://jira.percona.com/browse/PMM-9994): QAN: Improved the description of  the QAN filter to make it more clear.

- [PMM-9632](https://jira.percona.com/browse/PMM-9632): For flexibility, we have added the ability to change some of the pg_stat_monitor settings. Earlier it was hardcoded.

- [PMM-9921](https://jira.percona.com/browse/PMM-9921): Migrated pmm2-client Docker image to ubi-micro to reduce the number of packages and the image size.

- [PMM-10085](https://jira.percona.com/browse/PMM-10085): Support for recently released Percona Operators for MySQL (1.11.0)

- [PMM-5885](https://jira.percona.com/browse/PMM-5885): Upgraded the PostgreSQL version in the PMM server to 14.

> - [PMM-10122](https://jira.percona.com/browse/PMM-10122):

Reviewer 2: no description

## Bugs Fixed

- [PMM-10474](https://jira.percona.com/browse/PMM-10474): Fixed an issue where the PMM tour gets in the way when sharing a dashboard via a rendered image with an error message at the top (HTTP 404).

 

> - [PMM-9556](https://jira.percona.com/browse/PMM-9556): DBaaS: Fixed an issue where creating a MongoDB 5.0.x DB cluster in DBaaS was failing.

Reviewer 2: not fixed?

> - [PMM-10012](https://jira.percona.com/browse/PMM-10012): DBaaS: Fixed an issue where the creation of Percona Server for MongoDB (PSMDB) cluster failed with 1.12 operator.

Reviewer 2: not fixed

Author: @denisok Should I remove it then?

# Percona Monitoring and Management 2.30.0

| **Release date:** | Aug 22, 2022                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important"
    We recommend using the latest version of PMM. This ensures that you have access to the latest PMM features and that your environment runs on the latest version of the underlying components, such as VictoriaMetrics, with all the bug fixes in place.

## Release Highlights

#### New Experimental dashboards

!!! caution alert alert-warning "Important"
    These experimental dashboards aresubject to change. It is recommended to use these dashboards for testing purposes only.

> - We have introduced a more user-friendly and insightful experimental Home Dashboard in PMM version 2.30.0. With this new Home dashboard, you can experience a boost in performance while monitoring a large number of nodes or services in PMM. Repeated rows (for a large number of nodes) on the dashboard can take a considerable time to load the Home Dashboard UI. Our new experimental dashboard ensures fast retrieval of data on the Home dashboard.

Author: We can provide a link.

    - **Experimental MongoDB Collection Details**

        This experimental dashboard provides detailed information about the top collections by document count, size, and document read for MongoDB databases.

    - **Experimental MongoDB Oplog Details**

        This real-time dashboard contains Oplog details such as Recovery Window, Processing Time, Buffer Capacity, and Oplog Operations.

### Database as a Service (DBaaS)

Enhanced the user experience by simplifying the configuration and deployment of DBaaS database clusters:

- With PMM 2.30.0, you can create a DB cluster literally at the click of a button. All the fields will be automatically populated with the default value. Click Register, and your DB cluster is created.

- You can easily create a Kubernetes cluster with PMM 2.30.0. When you copy the value of the Kubeconfig file, the Kubernetes Cluster Name gets auto-populated from the contents of the kubeconfig file.

- Starting with PMM 2.23.0, we have simplified the DBaaS configuration and deployment of database clusters by providing suggestions on the k8s cluster name based on the Kubeconfig file.

### Operators support

> - PMM now supports Percona Operators MySQL (1.11) and MongoDB (1.12).

Reviewer 3: Aren't we using semver? Is it 1.11 or 1.11.0 ? Same for mongo

- PMM now provides MongoDB v5 deployment support. 

## New Features

> - [PMM-9652](https://jira.percona.com/browse/PMM-9652): Dashboard: New experimental dashboards for MongoDB are now available with more MongoDB metrics and data like Hottest collections, Data Size, recovery Window, Processing Time, Buffer Capacity, Oplog Operations, and so on.

Reviewer 3: case inconsistency: Hottest collections / Data Size / recovery Window ...

Author: Fixed it.

> - [PMM-10367](https://jira.percona.com/browse/PMM-10367): Dashboard: We have introduced a new experimental Home Dashboard in PMM version 2.30.0. The primary goal of this Home dashboard is to improve the performance while monitoring a large number of nodes or services in PMM. Repeated rows (for a large number of nodes) on the dashboard can take a considerable time to load Home Dashboard UI. So, to address this issue, we have developed a very user-friendly and insightful new experimental Home Dashboard.

Reviewer 3: Home dashboard / Home Dashboard

> - [PMM-10327](https://jira.percona.com/browse/PMM-10327):  You can create a free K8s cluster via Percona Platform Portal and use it for testing DBaaS in PMM.

Reviewer 3: ```suggestion
- [PMM-10327](https://jira.percona.com/browse/PMM-10327): You can create a free K8s cluster via Percona Platform Portal and use it for testing DBaaS in PMM.
```

Reviewer 3: Again: K8s, when previously we used k8s

    
    !!! note alert alert-primary ""

>         This feature will be available on https://portal.percona.com/ with the PMM 2.30 GA release.

Reviewer 3: ```suggestion
        This feature will be available on https://portal.percona.com/ with the PMM 2.30.0 GA release.
```

 
## Improvements

- [PMM-10340](https://jira.percona.com/browse/PMM-10340): DBaas: Enhanced user experience to simplify DBaaS configuration and deployment of Database clusters with suggestions on k8s cluster name based on k8s config file.
  

> - [PMM-10368](https://jira.percona.com/browse/PMM-10368): Simpler DBaaS and Percona Platform connection configuration by suggesting the public server address of the PMM server.

Reviewer 3: redundancy: "public server address of the PMM server" -> I guess "public address of the PMM server" or "public server address of PMM".

> - [PMM-5680](https://jira.percona.com/browse/PMM-5680): Logs: For an enhanced and simplified troubleshooting experience, we have added the following logs in a ZIP file in the ~/pmm-agent/ folder, which you can obtain from the pmm-admin-summary command:

Reviewer 3: should we backtick `pmm-admin-summary`?

    - nmysqld_exporter logs
    - mongodb_exporter logs
    - qan agent logs
    - pmm-agent log

  
- [PMM-5492](https://jira.percona.com/browse/PMM-5492): Logs: We have added the profiling data files to the logs folder to boost the troubleshooting process. You can obtain this by executing the `pmm-admin summary --pprof` command.

- [PMM-5005](https://jira.percona.com/browse/PMM-5005): Logs: Added a more pronounced error message for the PMM agent to help you distinguish between potential problems, such as connectivity issues, unconfigured agent issues, connection time-out issues, etc.

- [PMM-9968](https://jira.percona.com/browse/PMM-9968): Logs: You can now have more control over the logs you want to view with the `–log-level` parameter for the PMM agent. With this enhancement, you can experience more pronounced logging with the flexibility to choose options such as warning, error, and fatal as follows:

    `pmm-admin inventory add agent * --log-level=debug|info|warn|error|fatal`

> - [PMM-10147](https://jira.percona.com/browse/PMM-10147): Limited the number of jobs and actions to 32 that could be run concurrently by pmm-agent. That limit can be changed with the flag `--runner-capacity=<some number>` or environment variable `PMM_AGENT_RUNNER_CAPACITY=<some number>`. This feature helps to minimize resource consumption and the number of open DB connections.

Reviewer 3: ```suggestion
- [PMM-10147](https://jira.percona.com/browse/PMM-10147): Limited the number of jobs and actions to 32 that can be run concurrently by pmm-agent. That limit can be changed with the flag `--runner-capacity=<some number>` or environment variable `PMM_AGENT_RUNNER_CAPACITY=<some number>`. This feature helps to minimize resource consumption and the number of open DB connections.
```

- [PMM-10018](https://jira.percona.com/browse/PMM-10018): For optimum resource utilization when deploying a PXC DBaaS cluster, we have lowered the default CPU and Memory resources (CPU: 500m, memory: 500M).

  

> - [PMM-9914](https://jira.percona.com/browse/PMM-9914): PMM generates config files into the `/tmp` folder when a service is added to monitoring. Files in `/tmp` can get deleted by the Operating system. With PMM 2.30.0, we have ensured that the pmm-agent successfully restarts the child agents even though the config files in the /tmp folder are deleted and the scraper/exporter processes are killed thereby ensuring that the exporter is not broken.

Reviewer 3: ```suggestion
- [PMM-9914](https://jira.percona.com/browse/PMM-9914): PMM generates config files into the `/tmp` folder when a service is added to monitoring. Files in `/tmp` can get deleted by the operating system. With PMM 2.30.0, we have ensured that pmm-agent successfully restarts the child agents even though the config files in the `/tmp` folder are deleted and the scraper/exporter processes are killed, thereby ensuring that the exporter is not broken.
```

> - [PMM-9994](https://jira.percona.com/browse/PMM-9994): QAN: Improved the description of  the QAN filter to make it more clear.

Reviewer 3: ```suggestion
- [PMM-9994](https://jira.percona.com/browse/PMM-9994): QAN: Improved the description of the QAN filter to make it more clear.
```

- [PMM-9632](https://jira.percona.com/browse/PMM-9632): For flexibility, we have added the ability to change some of the pg_stat_monitor settings. Earlier it was hardcoded.

- [PMM-9921](https://jira.percona.com/browse/PMM-9921): Migrated pmm2-client Docker image to ubi-micro to reduce the number of packages and the image size.

- [PMM-10085](https://jira.percona.com/browse/PMM-10085): Support for recently released Percona Operators for MySQL (1.11.0)

- [PMM-5885](https://jira.percona.com/browse/PMM-5885): Upgraded the PostgreSQL version in the PMM server to 14.

- [PMM-10351](https://jira.percona.com/browse/PMM-10351): DBaaS: Full form of DBaaS (Database as a Service) is displayed on the Dashboard for clarity for the new users.  

> 

Reviewer 3: I see sometimes we leave 1 line between notes, other times 2, 3..

- [PMM-10122](https://jira.percona.com/browse/PMM-10122): Applied the pmm-agent `--log-level` to the exporter's output so that each message from the exporter with `log-level=WARN` will be logged in pmm-agent with the same `log-level=WARN`.

## Bugs Fixed

- [PMM-10474](https://jira.percona.com/browse/PMM-10474): Fixed an issue where the PMM tour gets in the way when sharing a dashboard via a rendered image with an error message at the top (HTTP 404).

 
- [PMM-9556](https://jira.percona.com/browse/PMM-9556): DBaaS: Fixed an issue where creating a MongoDB 5.0.x DB cluster in DBaaS was failing.

- [PMM-10012](https://jira.percona.com/browse/PMM-10012): DBaaS: Fixed an issue where the creation of Percona Server for MongoDB (PSMDB) cluster failed with 1.12 operator.

> - [PMM-10483](https://jira.percona.com/browse/PMM-10483): Fixed an issue where the QAN dashboard was not reloading after modifying the filters, refresh rate, sorting, etc

Reviewer 3: ```suggestion
- [PMM-10483](https://jira.percona.com/browse/PMM-10483): Fixed an issue where the QAN dashboard was not reloading after modifying the filters, refresh rate, sorting, etc.
```

> - [PMM-10299](https://jira.percona.com/browse/PMM-10299): Fixed an issue where the MongoDB ReplSet Dashboard overview section shows incorrect information for the Cluster and ReplSet members count and does not change even after  selecting a different cluster name.

Reviewer 3: ```suggestion
- [PMM-10299](https://jira.percona.com/browse/PMM-10299): Fixed an issue where the MongoDB ReplSet Dashboard overview section shows incorrect information for the Cluster and ReplSet members count and does not change even after selecting a different cluster name.
```

> - [PMM-10494](https://jira.percona.com/browse/PMM-10494): Fixed an issue where the Confirm action modal is stuck till the cluster is registered.

Reviewer 3: ```suggestion
- [PMM-10494](https://jira.percona.com/browse/PMM-10494): Fixed an issue where the Confirm action modal is stuck until the cluster is registered.
```

- We have simplified DBaaS and Percona Platform connection configuration by suggesting and setting the Public server address of the PMM server when a user connects to the Percona platform.

### Operators support

- PMM now supports Percona Operators MySQL (1.11) and MongoDB (1.12).

- PMM now provides MongoDB v5 deployment support. 

## New Features

- [PMM-9652](https://jira.percona.com/browse/PMM-9652): Dashboard: New experimental dashboards for MongoDB are now available with more MongoDB metrics and data like Hottest collections, Data Size, Recovery Window, Processing Time, Buffer Capacity, Oplog Operations, and so on.

- [PMM-10367](https://jira.percona.com/browse/PMM-10367): Dashboard: We have introduced a new user-friendly and insightful experimental Home Dashboard in PMM version 2.30.0. This Home dashboard improves the performance while monitoring a large number of nodes or services in PMM.

- [PMM-10327](https://jira.percona.com/browse/PMM-10327):  You can add a free K8s cluster via Percona Platform Portal and use it for testing DBaaS in PMM.
    
    !!! note alert alert-primary ""

>         This feature will be available on https://portal.percona.com/ with the PMM 2.30 GA release.

Reviewer 4: we don't, the point there is to specify that feature will be available on portal production at the time of PMM 2.30 release

>         This feature will be available on https://portal.percona.com/ with the PMM 2.30 GA release.

Reviewer 5: Maybe it's better to mention the date instead of the version then?

 
## Improvements

- [PMM-10340](https://jira.percona.com/browse/PMM-10340): DBaas: Enhanced user experience to simplify DBaaS configuration and deployment of Database clusters with suggestions on k8s cluster name based on k8s config file.
  
- [PMM-10368](https://jira.percona.com/browse/PMM-10368): Simpler DBaaS and Percona Platform connection configuration by suggesting the public server address of the PMM server.

- [PMM-5680](https://jira.percona.com/browse/PMM-5680): Logs: For an enhanced and simplified troubleshooting experience, we have added the following logs in a ZIP file in the ~/pmm-agent/ folder, which you can obtain from the pmm-admin-summary command:

    - nmysqld_exporter logs
    - mongodb_exporter logs
    - qan agent logs

>     - pmm-agent log

Author: ok.
